expo-maps/build/Map.types.d.ts

```diff
@@ -5,45 +5,146 @@ import { PolygonObject } from './Polygon';
+ */
 export declare type Gestures = {
     enableRotateGestures: boolean;
     enableScrollGestures: boolean;
     enableTiltGestures: boolean;
     enableZoomGestures: boolean;
 };
+/**
+ * Props for managing map controls settings.
+ *
+ * @googleMapsfield showZoomControls - if `true` zoom controls are visable
+ *
+ * @field showCompass - if `true` compass icon is visable
+ *
+ * @googleMapsfield showMapToolbar - if `true` map toolbar is visable
+ *
+ * @field showLevelPicker - if `true` level picker is visable when avaliable
+ */
 export declare type Controls = {
     showZoomControls: boolean;
     showCompass: boolean;
     showMapToolbar: boolean;
     showMyLocationButton: boolean;
     showLevelPicker: boolean;
 };
-export declare type GoogleMapsControls = Controls;
-export declare type AppleMapsControls = Omit<Controls, 'showMapToolbar' | 'showZoomControls'>;
+declare type GoogleMapsControls = Controls;
+declare type AppleMapsControls = Omit<Controls, 'showMapToolbar' | 'showZoomControls'>;
 export declare type NativeExpoGoogleMapsViewProps = ViewProps & PropsWithChildren<MapType & GoogleMapsStyling & Gestures & Markers & Polygons & Polylines & Circles & GoogleMapsControls>;
 export declare type NativeExpoAppleMapsViewProps = ViewProps & PropsWithChildren<MapType & Gestures & Markers & Polygons & Polylines & Circles & AppleMapsControls>;
 export declare type Providers = 'google' | 'apple';
+/**
+ * Prop for managing map provider.
+ *
+ * @field provider - provider you want to use for your map, please note `apple` provider is only avaliable on Apple devices,
+ * if `apple` provider is selected on Android device it is ignored
+ */
 export declare type Provider = {
     provider: Providers;
 };
+/**
+ * General Expo Map props.
+ *
+ * All of the ExpoMap props are optional, if prop is not explicitly specified default will be used.
+ *
+ * @field mapType - type of map (one of normal, hybrid, satellite, terrain')
+ * @default 'normal'
+ *
+ * @googleMapsfield googleMapsJsonStyleString - valid Google Maps style JSON string,
+ * please use https://mapstyle.withgoogle.com to generate style JSONs
+ * @default undefined
+ *
+ * @field enableRotateGestures - if `true` rotate gestures are enabled
+ * @default false
+ *
+ * @field enableScrollGestures - if `true` scroll gestures are enabled
+ * @default true
+ *
+ * @field enableTiltGestures - if `true` tilt gestures are enabled
+ * @default false
+ *
+ * @field enableZoomGestures - if `true` zoom gestures are enabled
+ * @default true
+ *
+ * TODO more explanations on how zoom controls function
+ * @googleMapsfield showZoomControls - if `true` zoom controls are visable
+ * @default true
+ *
+ * @field showCompass - if `true` compass icon is visable
+ * @default true
+ *
+ * @googleMapsfield showMapToolbar - if `true` map toolbar is visable
+ * @default true
+ *
+ * @field showLevelPicker - if `true` level picker is visable when avaliable
+ * @default true
```

Comment:
@ellwiii I know that these props can work conditionally. Could you help me to figure out these conditions?

Reply:
- for zoom controls they are visible at all times when set to "true",
- when compass is set to "true" it appears only when map is moved so that it is not facing north,
- tool bar shows when ta marker is tapped and hides when the marker is no longer in focus,
- level picker is available only when indoor maps are being used (for now we do not support this)

Reply:
Thanks!